std/assembly/internal/array.ts

```diff
@@ -36,14 +36,14 @@ export function weakHeapSort<T>(arr: Array<T>, comparator: (a: T, b: T) => i32):
   const typeShift = alignof<T>();
   const intShift  = alignof<i32>();
 
-  var blen = (len + 7) >> 3;
+  var blen = (len + 31) >> 5;
```

Comment:
Trying to understand this change. Before, there was one byte per 8 elements, if I understood this correctly:

8 elements -> blen=1
9 elements -> blen=2

Afterwards, with the change, this becomes:

8 elements -> blen=1
9 elements -> blen=1 (is this correct?)

Reply:
Right!

Before:
8 elements -> blen = 1
9 elements -> blen = 2

After:
32 elements -> blen = 1
33 elements -> blen = 2

Reply:
I might be missing something, but my impression was that the bitset needs one bit per element?

Reply:
Hmm, I'm not sure I quite understand your questions =)
 > So, if there are 3 elements, wouldn't it need 3 distinct bits still (one per element)?

Yes, 1 bit for each element.

> Otherwise, let's say there are three elements, the bitset would be 0011b, unable to distinguish this from element 1 and 2 being set?

It would be 0b011 or 0b111 or other combinations. So each bit can be set or not set.

Reply:
No every element has it's own independent bit position because it's like bit vector. Another word, we can represent our aux array like: ```new Array<bool>(len)``` in this case ```blen === len```. But this not optimal for memory usage, so we compress our bit array to 32-bit word

Reply:
You can read this for better understanding what I mean: https://nlogn.xyz/2018/01/17/bit-vector-in-javascript.html

Reply:
Maybe I'm just misunderstanding why an Uint32Array is used here, instead of a more natural Uint8Array.

Reply:
Because I hope Uint32Array is more performant. In JS version it's true

Reply:
Than an Uint8Array?

Reply:
If you mean Array<bool> or Array<u8> which store only one bit of 8-bits current version slower of course but we loose 7-bits and our memory usage is worst. So this is balance between memory usage/speed. I mean current version faster previous when I stored 8-bit vector inside 32-bit array =)